### ISSUE TITLE
standard_init_linux.go:211: exec user process caused "exec format error"

### ISSUE THREAD

**Issue by evertonevaristo** [kubernetes/autoscaler#3502]
I am trying to set up the cluster autoscaler, but unfortunately, I am getting the following error:

`standard_init_linux.go:211: exec user process caused "exec format error"`

All that I got in events logs its:

```
[ec2-user@ip-172-31-55-51 cluster-autoscaler]$ k get events -n kube-system | grep cluster-autoscaler | grep -v Normal | sort
21s         Warning   BackOff             pod/cluster-autoscaler-7b65cf79d5-9s7wg    Back-off restarting failed container
```

The cluster version: 

```
[ec2-user@ip-172-31-55-51 cluster-autoscaler]$ kubectl version
Client Version: version.Info{Major:"1", Minor:"15+", GitVersion:"v1.15.11-eks-065dce", GitCommit:"065dcecfcd2a91bd68a17ee0b5e895088430bd05", GitTreeState:"clean", BuildDate:"2020-07-16T02:00:53Z", GoVersion:"go1.12.17", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"15+", GitVersion:"v1.15.11-eks-065dce", GitCommit:"065dcecfcd2a91bd68a17ee0b5e895088430bd05", GitTreeState:"clean", BuildDate:"2020-07-16T01:44:47Z", GoVersion:"go1.12.17", Compiler:"gc", Platform:"linux/amd64"}
```

The image:

`cluster-autoscaler-7b65cf79d5-9s7wg:    us.gcr.io/k8s-artifacts-prod/autoscaling/cluster-autoscaler:v1.15.6,`

I already notice the migration of the google domain, but, this image it's available, and looks like it's compiled from a diferent architecutre...

**Comment by imyousuf** [comment]
I am facing the same issue in EKS -

Versions:

```bash
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.1", GitCommit:"c4d752765b3bbac2237bf87cf0b1c2e307844666", GitTreeState:"clean", BuildDate:"2020-12-23T02:22:53Z", GoVersion:"go1.15.6", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"18+", GitVersion:"v1.18.9-eks-d1db3c", GitCommit:"d1db3c46e55f95d6a7d3e5578689371318f95ff9", GitTreeState:"clean", BuildDate:"2020-10-20T22:18:07Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}
```

Images running -

```bash
$ kubectl get pods --all-namespaces -o jsonpath="{..image}" |\
> tr -s '[[:space:]]' '\n' |\
> sort |\
> uniq -c
      6 602401143452.dkr.ecr.us-east-1.amazonaws.com/amazon-k8s-cni-init:v1.7.5-eksbuild.1
      6 602401143452.dkr.ecr.us-east-1.amazonaws.com/amazon-k8s-cni:v1.7.5-eksbuild.1
      4 602401143452.dkr.ecr.us-east-1.amazonaws.com/eks/coredns:v1.7.0-eksbuild.1
      6 602401143452.dkr.ecr.us-east-1.amazonaws.com/eks/kube-proxy:v1.18.8-eksbuild.1
      2 us.gcr.io/k8s-artifacts-prod/autoscaling/cluster-autoscaler:v1.18.1
```

```bash
$ kubectl --namespace=kube-system logs -l "app.kubernetes.io/name=aws-cluster-autoscaler-chart"
standard_init_linux.go:219: exec user process caused: exec format error
```

**Comment by mariuskiessling** [comment]
You have to make sure that you are running the binary for your respective CPU architecture. In your case that is arm64. By default, cluster-autoscaler does not publish their images using mutli-arch images. Thus, you will have to select the correct image which for you would be `k8s.gcr.io/autoscaling/cluster-autoscaler-arm64:{version}`. If I see it correctly, there is only a pre-built ARM image available for cluster-autoscaler v1.20.0. Hence, you might have to build the image for the older version yourself.

**Comment by crazy-canux** [comment]
It's arm64.
